auth.go

```diff
@@ -4,99 +4,119 @@ import (
 	"bytes"
 	"crypto/sha256"
 	"errors"
-	log "github.com/sirupsen/logrus"
-	"go.mozilla.org/hawk"
 	"io"
 	"io/ioutil"
 	"mime"
 	"net/http"
 	"strings"
 	"time"
+
+	log "github.com/sirupsen/logrus"
+	"go.mozilla.org/hawk"
 )
 
-func auth(rf func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
+func auth(rf func(http.ResponseWriter, *http.Request), needsWrite bool) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		if !sruntime.cfg.Auth.DisableAuth {
 			hdr := r.Header.Get("Authorization")
-			v := false
+			v, wr := false, false
 			if strings.HasPrefix(hdr, "Hawk ") {
-				v = hawkAuth(r)
+				v, wr = hawkAuth(r)
 			} else if strings.HasPrefix(hdr, "APIKey ") {
-				v = apiAuth(r)
+				v, wr = apiAuth(r)
 			}
 			if !v {
 				w.WriteHeader(http.StatusUnauthorized)
 				return
 			}
+			if needsWrite && !wr {
+				w.WriteHeader(http.StatusForbidden)
```

Comment:
Should we just return the same status code across the board in here? Any reason this is different then above?

Reply:
At my last place we treated 401s as "you failed to auth" and 403s as "you're authed but dont have enough privs" -- I'm happy to make it a 401

Reply:
yeah lets just make the returned status codes from the authentication process consistent, there is minimal benefit in differentiating here right now